stmhal/dma.h

```diff
@@ -24,12 +24,59 @@
  * THE SOFTWARE.
  */
 
-//TODO: Put stream/channel defs for i2c/spi/can, etc here
+// These are ordered by DMAx_Stream number. The duplicate streams are ok
+// as long as they aren't used at the same time.
+//
+// Currently I2C and SPI are synchronous and they call dma_init/dma_deinit
+// around each transfer.
+
+#define DMA_STREAM_I2C1_RX      DMA1_Stream0
+#define DMA_CHANNEL_I2C1_RX                     DMA_CHANNEL_1
+
+#define DMA_STREAM_I2C2_RX      DMA1_Stream2
+#define DMA_CHANNEL_I2C2_RX                     DMA_CHANNEL_7
+
+#define DMA_STREAM_I2C3_RX      DMA1_Stream2
+#define DMA_CHANNEL_I2C3_RX                     DMA_CHANNEL_3
```

Comment:
Within the same DMA stream, should the definitions be ordered on channel number as well?  This means you don't have to shuffle lines around if/when you reallocate the channel.